src/foam/dao/jdbc/AbstractJDBCDAO.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.dao.AbstractDAO;
+
+import foam.core.*;
+
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.sql.*;
+
+import java.lang.Exception;
+import java.util.*;
+
+public abstract class AbstractJDBCDAO extends AbstractDAO{
+
+  /** Holds the relevant properties (column names) of the table */
+  protected List<PropertyInfo> properties_;
+
+  protected ThreadLocal<StringBuilder> threadLocalBuilder_ = new ThreadLocal<StringBuilder>(){
+
+    @Override
+    protected StringBuilder initialValue() {
+      return new StringBuilder();
+    }
+
+    @Override
+    public StringBuilder get() {
+      StringBuilder builder = super.get();
+      builder.setLength(0);
+      return builder;
+    }
+
+  };
+
+  protected String tableName_;
+
+  /** Holds a reference to the connection pool ( .getConnection() ) */
+  protected static DataSource dataSource_;
+
+  /**
+   * Create the table in the database and return true if it doesn't already exist otherwise it does nothing and returns false
+   * @param of ClassInfo
+   */
+  public abstract boolean createTable(X x, ClassInfo of);
+
+  public AbstractJDBCDAO(X x, ClassInfo of, String poolName) throws java.sql.SQLException, ClassNotFoundException {
+    setX(x);
+    setOf(of);
+
+    // Get the system global dataSource with its system global pool
+    JDBCPooledDataSource jp = new JDBCPooledDataSource(x, poolName);
+    dataSource_ = jp.getDataSource(x, poolName);
+
+    tableName_ = of.getObjClass().getSimpleName().toLowerCase();
+
+    getObjectProperties(of);
+
+    if ( ! createTable(x, of) ) {
```

Comment:
No need to check for error, since it will usually already be created. Maybe better to rename createTable() to maybeCreateTable() to indicate that it may or may not create the table.

Reply:
ok

Reply:
is "createTableIfNotCreated" too long?